packages/hoppscotch-common/src/components/collections/ImportExport.vue

```diff
@@ -1,361 +1,430 @@
+    title: "import.from_insomnia_description",
+    icon: IconInsomnia,
+    disabled: true,
+    applicableTo: ["personal-workspace", "url-import"],
+  },
+  component: FileSource({
+    caption: "import.from_file",
+    acceptedFileTypes: ".json",
+    onImportFromFile: async (content) => {
+      const res = await hoppInsomniaImporter(content)()
+
+      if (E.isRight(res)) {
+        onSuccessfulImport(res.right)
+      } else {
+        showImportFailedError()
+      }
+    },
+  }),
 }
 
-const onFileChange = () => {
-  stepResults.value = []
+const HoppGistImporter: ImporterOrExporter = {
+  metadata: {
+    id: "hopp_gist",
+    name: "import.from_gist",
+    title: "import.from_gist_description",
+    icon: IconGithub,
+    disabled: true,
+    applicableTo: ["personal-workspace", "url-import"],
+  },
+  component: GistSource({
+    caption: "import.from_url",
+    onImportFromGist: async (content) => {
+      const res = await hoppRESTImporter(content)()
 
-  const inputFileToImport = inputChooseFileToImportFrom.value[0]
+      if (E.isRight(res)) {
+        onSuccessfulImport(res.right)
+      } else {
+        showImportFailedError()
+      }
+    },
+  }),
+}
 
-  if (!inputFileToImport) {
-    hasFile.value = false
-    return
-  }
+const HoppMyCollectionsExporter: ImporterOrExporter = {
+  metadata: {
+    id: "hopp_my_collections",
+    name: "export.as_json",
+    title: "export.as_json_description",
+    icon: IconUser,
+    disabled: false,
+    applicableTo: ["personal-workspace"],
+    isLoading: isHoppMyCollectionExporterInProgress,
+  },
+  action: async () => {
+    isHoppMyCollectionExporterInProgress.value = true
 
-  if (!inputFileToImport.files || inputFileToImport.files.length === 0) {
-    inputChooseFileToImportFrom.value[0].value = ""
-    hasFile.value = false
-    toast.show(t("action.choose_file").toString())
-    return
-  }
+    const message = initializeDownloadCollection(
+      myCollectionsExporter(myCollections.value),
+      "Collections.json"
+    )
 
-  const reader = new FileReader()
+    if (E.isRight(message)) {
+      toast.success(message.right)
+    }
+
+    isHoppMyCollectionExporterInProgress.value = false
+  },
+}
+
+const HoppTeamCollectionsExporter: ImporterOrExporter = {
+  metadata: {
+    id: "hopp_team_collections",
+    name: "export.as_json",
+    title: "export.as_json_description",
+    icon: IconUser,
+    disabled: false,
+    applicableTo: ["team-workspace"],
+    isLoading: isHoppTeamCollectionExporterInProgress,
+  },
+  action: async () => {
+    isHoppTeamCollectionExporterInProgress.value = true
+
+    if (
+      props.collectionsType.type == "team-collections" &&
+      props.collectionsType.selectedTeam
+    ) {
+      const res = await teamCollectionsExporter(
+        props.collectionsType.selectedTeam.id
+      )
+
+      if (E.isRight(res)) {
+        initializeDownloadCollection(
+          res.right.exportCollectionsToJSON,
+          "team-collections.json"
```

Comment:
```suggestion
          "team-collections"
```
we do not need the .json here since its already set in the `initializeDownloadCollection` function.